website/addons/s3/static/s3UserConfig.js

```diff
@@ -0,0 +1,171 @@
+/**
+* Module that controls the S3 user settings. Includes Knockout view-model
+* for syncing data.
+*/
+
+var ko = require('knockout');
+require('knockout.punches');
+ko.punches.enableAll();
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+require('js/osfToggleHeight');
+
+var language = require('js/osfLanguage').Addons.s3;
+var osfHelpers = require('js/osfHelpers');
+var addonSettings = require('js/addonSettings');
+var ChangeMessageMixin = require('js/changeMessage');
+
+
+var ExternalAccount = addonSettings.ExternalAccount;
+
+var $modal = $('#s3InputCredentials');
+
+
+function ViewModel(url) {
+    var self = this;
+
+    self.properName = 'Amazon S3';
+    self.accessKey = ko.observable();
+    self.secretKey = ko.observable();
+    self.account_url = '/api/v1/settings/s3/accounts/';
+    self.accounts = ko.observableArray();
+
+    // Flashed messages
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('text-info');
```

Comment:
if you put `ChangeMessageMixin.call(self);` in the constructor, you don't have to define `self.message` and `self.messageClass` either.